mh/pat.py

```diff
@@ -2471,6 +2537,13 @@ def handle(self):
             self.server.error(traceback.format_exc())
             self.send_error("{}: {}".format(type(e).__name__, str(e)))
 
+        try:
+            circle_index = self.session.local_info['circle_id']
+            if circle_index is not None:
+                self.notify_circle_leave(circle_index, 0)
+        except Exception:
```

Comment:
I don't like this exception ignoring trap in this case. Since if a exception is caught here there is a cleaning logic error elsewhere and I rather be aware of those.

Reply:
I was thinking, "better safe than sorry," but you're right, this shouldn't be needed. Fixed.